encodings/fsst/src/array.rs

```diff
@@ -90,18 +111,36 @@ impl FSSTArray {
             .expect("Symbols must be a Primitive Array");
         let symbols = symbols_array.maybe_null_slice::<u64>();
 
+        let symbol_lengths_array = self
+            .symbol_lengths()
+            .into_canonical()
+            .unwrap()
+            .into_primitive()
+            .unwrap();
+        let symbol_lengths = symbol_lengths_array.maybe_null_slice::<u8>();
+
+        // SAFETY: we transmute to remove lifetime restrictions.
+        //  Without this, the compiler complains that `symbol_lengths is tied to the lifetime of
+        //  the `symbol_lengths_array` local variable, but it's actually tied to the lifetime of
+        //  the `symbols` child array of self. We can't represent this in the type system right now,
+        //  so we transmute to kill the lifetime complaints.
+        //  This is fine because the returned `Decompressor`'s lifetime is tied to the lifetime
+        //  of these same arrays.
+        let symbol_lengths = unsafe { std::mem::transmute::<&[u8], &[u8]>(symbol_lengths) };
```

Comment:
curious for a sanity check here, or if there's another way i should be doing this. it feels a bit wrong, but I think it is currently the best way to do the thing I want...

Reply:
nvm this is wrong, if we actually canonicalize this pointer is invalid

Reply:
ok this should be fixed now, instead of returning a decompressor this constructs one on-the-fly to pass to a provided function